fastar/core.py

```diff
@@ -131,8 +127,8 @@ def _compute_ancestral_child_counts(self, box):
         # TODO: pass var_idx to the backward rule
         raise NotImplementedError
       else:
-        instarts, counts = backward_rules[primitive](
-            to_global_coords(ubox), *invals, **params)
+        instarts, counts, _ = backward_rules[primitive](
+            to_global_coords(ubox), *(v.shape for v in invals), **params)
```

Comment:
As we discussed yesterday I think the backward rules should actually have access to the raw values, at least for non-LazyArray inputs. We can leave it like this (with just the shapes) for now and change back when we implement conv though.

Reply:
I will revert this but abstractify lazy cache values. This will combine the best of both approaches.

Reply:
Done!